lib/collection/src/shards/shard_holder.rs

```diff
@@ -155,6 +201,51 @@ impl ShardHolder {
         Ok(())
     }
 
+    pub fn start_resharding(
+        &mut self,
+        shard_id: ShardId,
+        shard: ShardReplicaSet,
+    ) -> Result<(), CollectionError> {
+        let Some(default_ring) = self.rings.get_mut(&None) else {
```

Comment:
Let's pass the sharding key in here right away, instead of defaulting to `None` here.

Reply:
There's no sharding key, because we don't support resharding custom sharding keys.

Personally, I would prefer to implement resharding for default sharding key first, and then extend it for custom sharding if necessary.

Reply:
> because we don't support resharding custom sharding keys.

Why?

It's essentially the same, but on a different scope.

Reply:
Because I did not implement it yet? :D

Also, I don't fully understand how custom sharding works (and if that would add any annoying corner cases to resharding),.

Reply:
> Also, I don't fully understand how custom sharding works

You can use any arbitrary key to create a different set of shards. 

In other words, each shard key, whether `None` or `Some(Key)` has their own set of shards. In all cases resharding would work exactly the same, but it does need to know about the current sharding key.

Reply:
Done.